lowdb.d.ts

```diff
@@ -1,530 +1,532 @@
+      iteratee: (num: number) => TResult
+    ): LoDashWrapper;
+
+    /**
+     * @see _.times
+     */
+    times(): LoDashWrapper;
+
+    /**
+     * @see _.uniqueId
+     */
+    uniqueId(): LoDashWrapper;
+
+    value<T>(): T;
+
+    pop<T>(): T;
+    push<T>(...items: T[]): LoDashWrapper;
+    shift<T>(): T;
+    sort<T>(compareFn?: (a: T, b: T) => number): LoDashWrapper;
+    splice<T>(start: number): LoDashWrapper;
+    splice<T>(start: number, deleteCount: number, ...items: any[]): LoDashWrapper;
+    unshift<T>(...items: T[]): LoDashWrapper;
+  }
+
+  interface Storage {
+    /**
+      * Reads the database.
+      *
+      * @param source The source location.
+      * @param deserialize The deserialize function to apply.
+      * @return Returns a promise with the deserialized db object.
+    */
+    read?(source: string, deserialize: any): PromiseLike<any>
+    /**
+      * Reads the database.
+      *
+      * @param source The source location.
+      * @param deserialize The deserialize function to apply.
+      * @return Returns the deserialized db object.
+    */
+    read?(source: string, deserialize: any): Object
+    /**
+      * Writes to the database.
+      *
+      * @param destination The destination location.
+      * @param obj The object to write.
+      * @param serialize The serialize function to apply.
+    */
+    write?(destination: string, obj: any, serialize: any): void
+    /**
+      * Writes to the database.
+      *
+      * @param destination The destination location.
+      * @param obj The object to write.
+      * @param serialize The serialize function to apply.
+    */
+    write?(destination: string, obj: any, serialize: any): PromiseLike<void>
+  }
+
+  interface Format {
+    /**
+      * Writes to the database.
+      *
+      * @param obj The object to serialize.
+      * @return Returns the serialized object string.
+    */
+    serialize(obj: Object): string
+    /**
+      * Writes to the database.
+      *
+      * @param data The object to deserialize.
+      * @return Returns the deserialized object.
+    */
+    deserialize(data: string): Object
+  }
+
+  interface Options {
+    /**
+     * The custom "storage" object.
+     */
+    storage?: Storage
+    /**
+     * The custom "format" object.
+     */
+    format?: Format
+    /**
+     * The flag to automatically persist changes.
+     */
+    writeOnChange?: boolean
+  }
+
+  interface Low extends LoDashWrapper, Format {
+    /**
+     * Access current database state.
+     * Returns Returns the database state.
+     */
+    getState(): Object
+    /**
+     * Drop or reset database state.
+     * @param newState New state of the database
+     */
+    setState(newState: Object): void
+    /**
+     * Persist database.
+     * @param source The source location.
+     */
+    write(source: string): void
+    /**
+     * Persist database.
+     * @param source The source location.
+     */
+    write(source: string): PromiseLike<any>
```

Comment:
No problem. Having this PR eventually merged, I could open up a follow-up PR (or, thanks to the introduced namespace, it is possible to augment the interface for the compiler from the outside).

The `README.md` [1] states it is `db.write([source])`, where `source` is optional.

~~~ts
const db = low('db.json')
db.write()            // writes to db.json
db.write('copy.json') // writes to copy.json
~~~

[1] https://github.com/typicode/lowdb/blob/v0.14.0/README.md#api